solidity/test/ReserveToken.js

```diff
@@ -0,0 +1,115 @@
+const { accounts, contract } = require('@openzeppelin/test-environment');
+const { expect } = require('../../chai-local');
+const { BN, balance } = require('@openzeppelin/test-helpers');
+
+const { NATIVE_TOKEN_ADDRESS } = require('./helpers/Constants');
+
+const ReserveToken = contract.fromArtifact('TestReserveToken');
+const TestStandardToken = contract.fromArtifact('TestStandardToken');
+
+const TOTAL_SUPPLY = new BN(1_000_000);
+
+describe('ReserveToken', () => {
+    const getBalance = async (reserveToken, account) => {
+        const reserveTokenAddress = reserveToken.address || reserveToken;
```

Comment:
We should really wrap our offchain reserveToken (or create a contract also for ETH reserve, and then change its `balanceOf`, `approve` and `address` hooks internally), so that we don't have this type ambiguity.

Reply:
Yes, we discussed it and we will do after Hardhat